### ISSUE TITLE
UX suggestions for the quick pick control

### ISSUE THREAD

**Issue by beastdestroyer** [microsoft/vscode#56076]
Issue Type: Feature Request/UI/UX
I haven't found any particular issues

The trusty activity bar. Down at the bottom of the window. VSCode couldn't live without it (probably). It also opens up a few buttons that open up panels, as I'll refer them to.

In this issue, I'd like to outline my suggestions for them, and these will probably apply to similar panels. The panels I'll be talking about are outlined in this paint document:
![panels](https://user-images.githubusercontent.com/30497573/43895419-26b1fba6-9bcd-11e8-9328-738b85d6d78f.png)
> I'm also using One Dark Pro theme

- End of Line Sequence: As there is nothing special about this, I can expand more on my idea for how these panels should look:
  - First of all, I believe everything should be bigger; in order to keep VSCode's, bigger on top and smaller on the bottom like Sublime, I propose the first box containing the search bar should be bigger and fatter,as it is the main item after all. (or at least the first one)
  - Secondly, the padding of items in the list should be increased. This makes it seem more user-friendly and will allow for more stuff to be added (as will be seen later on). I only seem to see line-height modifying it (from file explorer anyways), how about a general padding of let's say 10px? (I always like padding in multiples of 5).
  - Thirdly, font-size should increase. I'm not sure of the actual font-size, but it seems to me about around 13px, I can't developer tools it since it'll just close when I try it, so I took it from file explorer instead. Perhaps bumping it a bit to 14px or 16px could do something.
  - Fourthly, Just like Atom and Sublime, it looks nice when the max width is the same as the searchbar, but really this is more of an opinion, I don't mind how it looks right now being full-width
  - I also do understand that in order to look nice, line height has to increase too, but looking at the padding that would be added, a boost wouldn't really need to be that high
  - With regards to a talk about File Explorer and similar menus, could it be possible that these changes reflect it? I'm sure this would break the look of many other menus which have a similar style, such as the Settings GUI and just about everything on the sidebar. The title tags would have to rescale too.

- Next up is the Language Mode Panel!
  - Changes included above, but of course the icons would have to be bigger. (Unless the font-size stays the same, and padding isn't too much) and definitely will be a burden to icon theme devs, especially huge ones such as the popular ```vscode-icons``` which I use. If the files uses SVG (like vscode-icons), it wouldn't be that much of a problem since it scales. But if not.. well I'm sorry :)
- Now, the Select Action Panel:
  - Above changes in no. 1, plus the blue text could be under the main title, maybe similar to the Settings GUI Layout perhaps? I;m sure these blue text are like mini descriptors of that it will do. Perhaps take inspiration from that? 
  - If it definitely is an individual sub-action, then perhaps consider leaving it where it is, but turn it into a visible  button of some sort
- Finally, the File Encoding panel
  - All of no. 1 plus, again taking inspiration from Settings GUI, making it more faded in colour. This could be applied to the subscripted/smaller text in the Language Mode panel too

Basically it, I wanted to give my thoughts. I can probably imagine the disagreements too. Have a great day!

**Comment by beastdestroyer** [comment]
Another thing, added screenshots from Atom to compare: (Sublime Text doesn't seem to have any nice GUI for these :( )
1. Language Mode Panel:
Atom:
![image](https://user-images.githubusercontent.com/30497573/43897722-de4aa2a2-9bd4-11e8-8cb0-fc4df2520cf6.png)
2. End of Line Panel:
Atom:
![image](https://user-images.githubusercontent.com/30497573/43897756-fb689c36-9bd4-11e8-9614-a1a186dddfee.png)
3. Character Encoding:
Atom: 
![image](https://user-images.githubusercontent.com/30497573/43897800-330ce2c8-9bd5-11e8-9463-de9e93cab38c.png)
4. Actions not applicable

**Comment by beastdestroyer** [comment]
Another thing: The 4 items that appear when a language is selected: "Auto Detect", "Configure File Association for 'xxxxx' ", "Configure 'xxxxx' language based settings" and "Search Marketplace Extensions for 'xxxxx' ", applies to these when they appear.

**Comment by ramya-rao-a** [comment]
@beastdestroyer Thanks for your input. I wanted to clarify a few things based on your request above.

> The trusty activity bar. Down at the bottom of the window. VSCode couldn't live without it (probably). It also opens up a few buttons that open up panels, as I'll refer them to.

That bar is called the status bar. We refer to the vertical bar that has the icons/buttons for file explorer, scm viewlet, search, extensions and debug viewlet as the activity bar. I am just saying this to avoid confusion in the further discussions.

The panels you are referring to are quick pick controls. They get used by extensions as well whenever there is a feature that needs the user to choose from multiple choices.

> Now, the Select Action Panel:
Above changes in no. 1, plus the blue text could be under the main title, maybe similar to the Settings GUI Layout perhaps? I;m sure these blue text are like mini descriptors of that it will do. Perhaps take inspiration from that?
If it definitely is an individual sub-action, then perhaps consider leaving it where it is, but turn it into a visible button of some sort

The blue text is not a mini description. Its a call to action. We wont be moving it. The idea of having a button is nice but only if we increase the height of the line. I'll leave that to @misolori to address.

> Finally, the File Encoding panel
All of no. 1 plus, again taking inspiration from Settings GUI, making it more faded in colour. This could be applied to the subscripted/smaller text in the Language Mode panel too

We will not be having different UI treatments for the different instances of the same control. So we cannot take special requests around specific cases like this.

To summarize, the only actionable items from your request are
- Increase the line height/padding around each item in the quick pick control
- Consider having the links on the right as buttons

**Comment by V-ed** [comment]
> Increase the line height/padding around each item in the quick pick control

The little padding around each item in the quick pick control is something that bothered me from day one when I started using VSCode and still irks me from time to time. I got accustomed to it but I'd really like to be able to go toward bigger padding, as I came from sublime and atom and prefer their quick pick control.

Looking forward to maybe have a customization option to tweak such padding!
